reset dynamo cache before each test

shunting314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126586

In https://github.com/pytorch/pytorch/issues/125967, we found test results depend on test order. The root cause is due to earlier tests populate dynamo cache and affect the later tests.

This PR clear dynamo cache before each unit test so we get more deterministic result for unit test

shunting314 committed:
reset dynamo cache before each test

[ghstack-poisoned]

huydhn commented on torch/testing/_internal/common_utils.py:
```diff
@@ -2903,6 +2903,11 @@ def setUp(self):
+        # reset dynamo cache to avoid issues like
+        # https://github.com/pytorch/pytorch/issues/125967#issuecomment-2118483919
+        # which depends on test order.
+        torch._dynamo.reset()
```
I vaguely remember that this causes a perf hit on CI, so it couldn't be rolled out before.  @zou3519 would know more

zou3519 replied:
We do need to do this (reset dynamo cache). I'm not sure about the performance impact on inductor tests, but it is hopefully not bad because we do not run that many inductor tests.

@shunting314 can you get the tests to pass in this PR so that we can see the before/after test timing numbers?

shunting314 replied:
yea. 2 of the failures are due to previously fail tests now pass. I removed the file marking the test failure.

For the other failures, I'm not able to repro. E.g., for https://github.com/pytorch/pytorch/actions/runs/9135411752/job/25122927855 , I run 
```
PYTORCH_TEST_WITH_INDUCTOR=1 python test/functorch/test_ops.py -k test_extremal_numerics_l1_loss_cpu
```
And it passes on my devserver..

Any suggestions for this?

zou3519 replied:
it might have to do with python version. Try to skip these for now. Also, the `keep-going` label will make all tests run -- it's possible you have more failures than just these two

shunting314 replied:
Since the PR is clean now (the previous failed tests shows up as flaky and non-blocking in CI), I'll merge it.

zou3519 replied:
> Since the PR is clean now (the previous failed tests shows up as flaky and non-blocking in CI), I'll merge it.

Yup go for it

shunting314 committed:
Update on "reset dynamo cache before each test"


In https://github.com/pytorch/pytorch/issues/125967, we found test results depend on test order. The root cause is due to earlier tests populate dynamo cache and affect the later tests.

This PR clear dynamo cache before each unit test so we get more deterministic result for unit test



[ghstack-poisoned]

shunting314 commented:
@pytorchbot merge

shunting314 commented:
hmm, the test failures blocks the merge even though CI is green before I try to merge.

I'll skip those tests for now.

shunting314 committed:
Update on "reset dynamo cache before each test"


In https://github.com/pytorch/pytorch/issues/125967, we found test results depend on test order. The root cause is due to earlier tests populate dynamo cache and affect the later tests.

This PR clear dynamo cache before each unit test so we get more deterministic result for unit test



[ghstack-poisoned]

clee2000 commented:
@pytorchbot revert -m "broke tests on inductor? test_modules.py::TestModuleCUDA::test_cpu_gpu_parity_nn_CTCLoss_cuda_float64 https://hud.pytorch.org/pytorch/pytorch/commit/43f2f43eb3b6d8cbe8eb7f45acb50376092f1a16 https://github.com/pytorch/pytorch/actions/runs/9200644034/job/25308511495" -c nosignal

shunting314 committed:
Update on "reset dynamo cache before each test"


In https://github.com/pytorch/pytorch/issues/125967, we found test results depend on test order. The root cause is due to earlier tests populate dynamo cache and affect the later tests.

This PR clear dynamo cache before each unit test so we get more deterministic result for unit test



[ghstack-poisoned]

shunting314 committed:
Update on "reset dynamo cache before each test"


In https://github.com/pytorch/pytorch/issues/125967, we found test results depend on test order. The root cause is due to earlier tests populate dynamo cache and affect the later tests.

This PR clear dynamo cache before each unit test so we get more deterministic result for unit test



[ghstack-poisoned]

shunting314 committed:
Update on "reset dynamo cache before each test"


In https://github.com/pytorch/pytorch/issues/125967, we found test results depend on test order. The root cause is due to earlier tests populate dynamo cache and affect the later tests.

This PR clear dynamo cache before each unit test so we get more deterministic result for unit test



[ghstack-poisoned]

shunting314 committed:
Update on "reset dynamo cache before each test"


In https://github.com/pytorch/pytorch/issues/125967, we found test results depend on test order. The root cause is due to earlier tests populate dynamo cache and affect the later tests.

This PR clear dynamo cache before each unit test so we get more deterministic result for unit test



[ghstack-poisoned]

shunting314 commented:
@pytorchbot merge

huydhn commented:
@pytorchbot drci

huydhn commented:
@pytorchbot merge

malfet commented:
@pytorchbot revert -m "Broke tons of tests, see https://hud.pytorch.org/pytorch/pytorch/commit/bd24991f461476036d6ba20fed92651c7e46ef7c " -c ignoredsignal
